### ISSUE TITLE
Batching persistence journal will never emit any failed operation messages

### ISSUE THREAD

**Issue by Arkatufus** [akkadotnet/akka.net#4951]
The way it is implemented right now, batch execution is done through the `ExecuteChunk` method and is guarded by a circuit breaker that will emit a `ChunkExecutionFailure` that in turn will report a failure message to all requesters.
The problem with the current implementation is that the `ExecuteChunk` body is guarded with a try...catch that consumes all exceptions and always returns a `BatchComplete` message that returns a possible cause of failure, making the circuit breaker practically useless.

Possible solutions would be:
  1. Fix the code so that the circuit breaker does it work:
      - Make sure the try...catch block re-throw the exception so that it can be caught by the circuit breaker, which then will run the proper error reporting for each requests in the batch.
      - Remove the useless and redundant `Cause` property from the `BatchComplete` message because, now, it will only fire if the batch completed successfully.
  2. Fix the batch result handler:
      - Remove the useless circuit breaker 
      - Harden the `BatchComplete` handler by moving the `ChunkExecutionFailure` handler logic into the `CompleteBatch` method
      - Remove the now useless `ChunkExecutionFailure` message and handler because it will never be fired.

**Comment by Aaronontheweb** [comment]
Is this why https://github.com/akkadotnet/Akka.Persistence.SqlServer/issues/114#issuecomment-496951433 is still an issue possibly?

**Comment by Aaronontheweb** [comment]
cc @ismaelhamed

**Comment by Arkatufus** [comment]
yes, that is exactly the cause of that issue, because it will never be called.

**Comment by Aaronontheweb** [comment]
@Arkatufus this has been fixed now, correct?
